index.js

```diff
@@ -32,41 +30,56 @@ exports.kinesisHandler = (records, opts = {}, context) => {
       });
     }
 
-    // Required parameters are valid execute the following:
-    // 1) Obtain the decoded kinesis data
-    // 2) Obtain a valid OAuth token to process record data
     const schema = opts.schema;
     const apiUri = opts.apiUri;
+    const hrcModel = new HoldRequestConsumerModel();
     const streamsClient = new NyplStreamsClient({ nyplDataApiClientBase: apiUri });
     const apiHelper = new ApiServiceHelper(
       process.env.OAUTH_PROVIDER_URL,
       process.env.OAUTH_CLIENT_ID,
       process.env.OAUTH_CLIENT_SECRET,
       process.env.OAUTH_PROVIDER_SCOPE
     );
-    const holdRequestConsumerModel = new HoldRequestConsumerModel();
 
     Promise.all([
       apiHelper.getOAuthToken(CACHE['access_token']),
       streamsClient.decodeData(schema, records.map(i => i.kinesis.data))
     ]).then(result => {
-      // The access_token has been obtained and all records have been grouped by source
-      // Next, we need create a string from the record and nyplSource keys to perform a GET request
-      // to the ItemService
       CACHE['access_token'] = result[0];
```

Comment:
Are we always requesting for access token? Or saving it once we get it and reuse it until it expires?